From 1ab9a805b13959f1e8702dab919925713643500d Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Mon, 27 Apr 2020 21:51:57 -0400
Subject: [PATCH] Suppress dependency versions which are known to be too
 pessimistic

These all work just fine in reported use and should not be pinned; we
package versions in Arch which are "too new".

Also fix completely broken dependency pinning which tries to enforce
minimum versions of dependencies, but pins it twice, once with more
relaxed pinning for older versions of python. If an old version works,
it works irrespective of the python version, and you don't need to use
minimum requirements as your mallet to force people to upgrade to the
latest version anyway.
https://github.com/dephell/dephell/issues/330
---
 pyproject.toml | 16 +++++-----------
 1 file changed, 5 insertions(+), 11 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 7a38469..152e7af 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -27,15 +27,15 @@ cleo = "^0.7.6"
 clikit = "^0.4.2"
 requests = "^2.18"
 cachy = "^0.3.0"
-requests-toolbelt = "^0.8.0"
+requests-toolbelt = "*"
 jsonschema = "^3.1"
-pyrsistent = "^0.14.2"
+pyrsistent = "*"
 pyparsing = "^2.2"
 cachecontrol = { version = "^0.12.4", extras = ["filecache"] }
 pkginfo = "^1.4"
 html5lib = "^1.0"
 shellingham = "^1.1"
-tomlkit = "^0.5.11"
+tomlkit = "*"
 pexpect = "^4.7.0"
 
 # The typing module is not in the stdlib in Python 2.7 and 3.4
@@ -49,19 +49,13 @@ glob2 = { version = "^0.6", python = "~2.7 || ~3.4" }
 virtualenv = { version = "^16.7.9", python = "~2.7" }
 # functools32 is needed for Python 2.7
 functools32 = { version = "^3.2.3", python = "~2.7" }
-keyring = [
-    { version = "^18.0.1", python = "~2.7 || ~3.4" },
-    { version = "^20.0.1", python = "^3.5" }
-]
+keyring = "*"
 # Use subprocess32 for Python 2.7 and 3.4
 subprocess32 = { version = "^3.5", python = "~2.7 || ~3.4" }
 importlib-metadata = {version = "~1.1.3", python = "<3.8"}
 
 [tool.poetry.dev-dependencies]
-pytest = [
-    {version = "^4.1", python = "<3.5"},
-    {version = "^5.4.3", python = ">=3.5"}
-]
+pytest = { version = ">=4.1" }
 pytest-cov = "^2.5"
 mkdocs = { version = "^1.0", python = "~2.7.9 || ^3.4" }
 pymdown-extensions = "^6.0"
-- 
2.27.0

